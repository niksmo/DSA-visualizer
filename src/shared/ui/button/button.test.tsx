import { render } from '@testing-library/react';
import { Button } from '.';
import userEvent from '@testing-library/user-event';

describe('Button component test', () => {
  it('should render with text', () => {
    const { getByRole } = render(<Button text="clickMe" />);
    const button = getByRole('button');
    expect(button).toMatchSnapshot();
  });

  it('should render without text', () => {
    const { getByRole } = render(<Button />);
    const button = getByRole('button');
    expect(button).toMatchSnapshot();
  });

  it('should render disabled state', () => {
    const { getByRole } = render(<Button disabled />);
    const button = getByRole('button');
    expect(button).toMatchSnapshot();
  });

  it('should render loading state', () => {
    const { getByRole } = render(<Button isLoader />);
    const button = getByRole('button');
    expect(button).toMatchSnapshot();
  });

  it('should click', () => {
    const click = jest.fn();
    const { getByRole } = render(<Button onClick={click} />);

    const button = getByRole('button');

    expect(click).toBeCalledTimes(0);

    userEvent.click(button);

    expect(click).toHaveBeenCalledTimes(1);
  });
});
