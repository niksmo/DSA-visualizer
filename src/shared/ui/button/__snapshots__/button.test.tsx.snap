// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test should render disabled state 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component test should render loading state 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component test should render with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    clickMe
  </p>
</button>
`;

exports[`Button component test should render without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
